onnxruntime/contrib_ops/cpu/word_conv_embedding.cc

```diff
@@ -160,6 +161,9 @@ Status WordConvEmbedding::ValidateInputShape(const TensorShape& w_conv_shape, co
 }
 
 Status WordConvEmbedding::Compute(OpKernelContext* ctx) const {
+  auto ctx_internal = static_cast<OpKernelContextInternal*>(ctx);
+  auto tp = ctx_internal->GetOperatorThreadPool();
```

Comment:
>auto [](start = 2, length = 4)

it's clearer to use 'const', '*' and '&' with auto where appropriate

Reply:
I copy-pasted it from existing code to keep consistent. We should do a clean on all such code to remove the casting to XXXinternal